services/provider/server/server_test.go

```diff
@@ -24,9 +26,9 @@ import (
 )
 
 type externalResource struct {
-	Kind string            `json:"kind"`
-	Data map[string]string `json:"data"`
-	Name string            `json:"name"`
+	Kind string `json:"kind"`
+	Data []byte `json:"data"`
```

Comment:
you are replacing one dependency w/ another, in test this could be `Data any json:"data"` as the mustMarshal is already a generic function and the result is `[]byte` that can be used w/o any conversion in protobuf.